build.rs

```diff
@@ -1,26 +1,213 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use std::path::PathBuf;
+use std::{
+    env, fs, io,
+    path::{Path, PathBuf},
+    pin::pin,
+};
 
+use anyhow::Context;
+use async_compression::futures::write::ZstdEncoder;
+use cid::Cid;
+use futures::{
+    io::{BufReader, BufWriter},
+    AsyncRead, AsyncWriteExt, Stream, StreamExt, TryStreamExt,
+};
+use fvm_ipld_car::{CarHeader, CarReader};
+use itertools::Itertools;
+use once_cell::sync::Lazy;
 use protobuf_codegen::Customize;
+use tokio::task::JoinSet;
 use walkdir::WalkDir;
 
 const PROTO_DIR: &str = "proto";
 const CARGO_OUT_DIR: &str = "proto";
 
-fn main() -> anyhow::Result<()> {
+const ACTOR_BUNDLE_CACHE_DIR: &str = "target/actor_bundles/";
```

Comment:
It'd be great to have it as a `PathBuf`. The less we operate on strings and more on strongly typed objects, the less likely we'll introduce a bug.

Reply:
Fixed